--- kcc-5.6.3/kindlecomicconverter/KCC_gui.py	2023-08-12 03:00:30.000000000 +0800
+++ kcc-5.6.3-b/kindlecomicconverter/KCC_gui.py	2023-09-20 21:06:11.833151566 +0800
@@ -43,11 +43,17 @@
 from . import KCC_ui_editor
 
 
+HOME = os.path.expanduser("~")
+XDG_DATA_HOME = os.environ.get("XDG_DATA_HOME", os.path.join(HOME, ".local", "share"))
+
+
+
 class QApplicationMessaging(QtWidgets.QApplication):
     messageFromOtherInstance = QtCore.pyqtSignal(bytes)
 
     def __init__(self, argv):
         QtWidgets.QApplication.__init__(self, argv)
+        self.setDesktopFileName("io.github.ciromattia.kcc.desktop");
         self._key = 'KCC'
         self._timeout = 1000
         self._locked = False
@@ -836,7 +842,7 @@
     def detectKindleGen(self, startup=False):
         if not sys.platform.startswith('win'):
             try:
-                os.chmod('/usr/local/bin/kindlegen', 0o755)
+                os.chmod(f'{XDG_DATA_HOME}/kindlegen', 0o755)
             except Exception:
                 pass
         kindleGenExitCode = Popen('kindlegen -locale en', stdout=PIPE, stderr=STDOUT, stdin=PIPE, shell=True)
