--- kcc-7.2.0/kindlecomicconverter/KCC_gui.py	2025-02-10 11:10:56.000000000 +0800
+++ kcc-7.2.0/kindlecomicconverter/KCC_gui2.py	2025-02-19 21:04:10.892195147 +0800
@@ -46,11 +46,13 @@
 from . import KCC_ui_editor
 
 
+
 class QApplicationMessaging(QApplication):
     messageFromOtherInstance = Signal(bytes)
 
     def __init__(self, argv):
         QApplication.__init__(self, argv)
+        self.setDesktopFileName("io.github.ciromattia.kcc");
         self._key = 'KCC'
         self._timeout = 1000
         self._locked = False
